Clear info on flatpak wrapper, including feature differences from original app (#133)
flathub/com.discordapp.Discord: README.md

# Discord

Discord is a free all-in-one messaging, voice and video client that's available on your computer and phone.

This repo hosts the flatpak wrapper for [Discord](https://discord.com/), available at [Flathub](https://flathub.org/apps/details/com.discordapp.Discord).

```sh
flatpak install flathub com.discordapp.Discord
flatpak run com.discordapp.Discord
```

## Differences in flatpak version

> The flatpak version runs in a sandbox to provide better safety and privacy for users (see [this privacy report](https://spyware.neocities.org/articles/discord.html)).

Reviewer: This feels inappropriate to put in a page which is supposed to represent the application.

Author: ```suggestion
The flatpak version runs in a sandbox to provide better safety and privacy for users.
```

However, this sandboxing prevents the following features from working:

- **Game Activity**: This flatpak version of Discord cannot scan running processes to detect running games.  
  There is currently no workaround or solution for this limitation.

> - **Drag and Drop Files**: To work around this now, you can change sandbox permissions of installed flatpak applications to give Discord file system access (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) or with `flatpak override --user --filesystem=home:ro com.discordapp.Discord`).

Reviewer 2: ```suggestion
- **Drag and Drop Files**: To work around this now, you can change sandbox permissions of installed flatpak applications to give Discord file system access:

```sh
flatpak override --filesystem=home:ro com.discordapp.Discord```
```

---

At line `8` you typed `flatpak install flathub com.discordapp.Discord` (without the `--user` flag), so when needing to change the permissions, you can't use `--user` since you installed it in your system.

IMO while I'm a big fan of Flatseal, I don't think instructions with Flatseal should be included in READMEs, or at least make it look better by doing a step-by-step guide rather than tell the user they can use Flatseal with no further instructions. It clutters a lot and makes it hard to follow through the README.

Author: > you can't use --user since you installed it in your system.

@TheEvilSkeleton I just tested this on my system installation of discord flatpak and yes it's working with --user override. See my comments above. https://github.com/flathub/com.discordapp.Discord/pull/133#discussion_r552984244

Author: Without --user, I need sudo to override

Reviewer 2: Wtf? I just tested it in a VM and it worked. Forget what I've said :P

Author: flatpak docs is definetly confusing

Reviewer 2: I updated the suggestion so it should look better.

Reviewer 2: > flatpak docs is definetly confusing

Agreed

Author: "allow edits by mantainers" is on, so you guys can commit whatever you want too, I think

Reviewer 2: I'm not a maintainer, just a user lurking around.

Author: Ah, ok. I'm a noob doing this, so I don't know how this all works

Reviewer 2: Pretty sure you can press on "Accept changes", or something like that right?

Author: yeah, I'll add that with the other link thing
Do you know if I can just add <a href=... on the appdata.xml?

Reviewer 2: No idea.

> - **Unrestricted File Access**: Default sandbox permissions limit Discord to only certain directories, so you can't access your entire Home directory. This limits which file directories you can attach files from and impacts drag and drop functionality. You can change sandbox permissions of installed flatpak applications (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal)) to give Discord broader file system access, allowing file attachments from more locations.

Reviewer: I would mention here that the plan is for Zypak to give us a file picker portal which will allow full access to the filesystem while still restricting unauthorized access.

Author: What would be the flatpak override command to allow access to Home? Is it `flatpak override --filesystem=home com.discordapp.Discord` ? Should I add this command as an alternative to Flatseal? @lionirdeadman

Author: ```suggestion
- **Unrestricted File Access**: Default sandbox permissions limit Discord to only certain directories, so you can't access your entire Home directory. Currently, this limits which file directories you can attach files from and impacts drag and drop functionality.  
  This limitation will likely be overcomed eventually, when Zypak give us a file picker portal which will allow full access to the filesystem while still restricting unauthorized access.  
  To work around this now, you can change sandbox permissions of installed flatpak applications (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) or with `flatpak override --filesystem=home com.discordapp.Discord`) to give Discord broader file system access, allowing file attachments from more locations.
```

Reviewer: It needs --user if it's installed as user application rather than system.

Author: Well it's just an example anyways. It might also need sudo I think, if it's installed on system, but you'd get an error and know what to do. Feel free to rephrase it if needed.

Author: Oh, okay. I didn't know you could add --user overrides on system installed apps too. I'll change it then.

Reviewer: You can't add --user to system installed apps. System and user applications have different permission overrides. Confusing, I know.

Author: Are you sure? I just tested with a system installed Discord flatpak (confirmed with `flatpak info com.discordapp.Discord`). I did `flatpak override --user ...`, `flatpak info --show-permissions ...` and then again with `su otheruser` and it does show the overridden home permission for just one user.

Author: Though maybe we should change it to `home:ro` if the purpose is only add attachments.
I didn't test this in the actual app by the way, whether just adding the `home` or `home:ro` permission actually allows file attachments properly. I'm assuming this works by what you guys have discussed.

Reviewer: Your flatpak info is for the system one, it also needs to be specificed --user.

`home:ro` would definitely work but I think realistically that zypak should get out of beta probably soon so it's not a mess I'm ready to make.

Author: That's the thing, there's no user installation (`flatpak info --user ...` fails, with and without --show-permissions), I only have the default system installation for Discord.
Yet the --show-permissions shows the home permission only on the user that added it (with override --user). I haven't tested it, but this makes it seem that you **can** override it per user even on system wide installation. Either this is documented confusingly [here](https://docs.flatpak.org/en/latest/flatpak-command-reference.html#flatpak-override), or maybe it's some kind of bug...?

# Discord

Discord is a free all-in-one messaging, voice and video client that's available on your computer and phone.

This repo hosts the flatpak wrapper for [Discord](https://discord.com/), available at [Flathub](https://flathub.org/apps/details/com.discordapp.Discord).

```sh
flatpak install flathub com.discordapp.Discord
flatpak run com.discordapp.Discord
```

## Differences in flatpak version

The flatpak version runs in a sandbox to provide better safety and privacy for users.

However, this sandboxing prevents the following features from working:

- **Game Activity**: This flatpak version of Discord cannot scan running processes to detect running games.  
  There is currently no workaround or solution for this limitation.
- **Unrestricted File Access**: Default sandbox permissions limit Discord to only certain directories, so you can't access your entire Home directory. Currently, this limits which file directories you can attach files from and impacts drag and drop functionality.  
  This limitation will likely be overcomed eventually, when Zypak give us a file picker portal which will allow full access to the filesystem while still restricting unauthorized access.  

>   To work around this now, you can change sandbox permissions of installed flatpak applications (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) or with `flatpak override --filesystem=home com.discordapp.Discord`) to give Discord broader file system access, allowing file attachments from more locations.

Author: ```suggestion
  To work around this now, you can change sandbox permissions of installed flatpak applications (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) or with `flatpak override --user --filesystem=home com.discordapp.Discord`) to give Discord broader file system access, allowing file attachments from more locations.
```

> - **Unrestricted File Access**: Default sandbox permissions limit Discord to only certain directories, so you can't access your entire Home directory. Currently, this limits which file directories you can attach files from and impacts drag and drop functionality.  
>   This limitation will likely be overcomed eventually, when Zypak give us a file picker portal which will allow full access to the filesystem while still restricting unauthorized access.  
>   To work around this now, you can change sandbox permissions of installed flatpak applications (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) or with `flatpak override --user --filesystem=home com.discordapp.Discord`) to give Discord broader file system access, allowing file attachments from more locations.

Author: ```suggestion
- **Drag and Drop Files**: To work around this now, you can change sandbox permissions of installed flatpak applications to give Discord file system access (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) or with `flatpak override --user --filesystem=home:ro com.discordapp.Discord`).
```

Reviewer: This seems good to me. Thanks for being fast and doing this btw! I'll merge this when #132 gets merged then.
